tests/lib/rules/array-callback-return.js

```diff
@@ -82,6 +142,51 @@ ruleTester.run("array-callback-return", rule, {
         { code: "foo.map(function* () {})", parserOptions: { ecmaVersion: 6 } }
     ],
     invalid: [
+
+        // options: { allowImplicit: false, checkForEach: true }
```

Comment:
Similar to the comment for docs, these options are completely unrelated. 

I think it's fine to have a couple of tests to check whether these options accidentally affect each other, but in general they should be tested separately.